packages/web3-shared/src/utils/query.ts

```diff
@@ -0,0 +1,20 @@
+import type { PastLogsOptions, Log } from 'web3-core'
+import type { ChainId } from '../types'
+import { getRPCConstants } from '../constants'
+
+// Metamask RPC is not reliable for method 'eth_getLogs', since it is set casually by the user.
+//  So we use our own default controllable RPC.
+export async function getPastLogs(queryParam: PastLogsOptions, chainId: ChainId): Promise<Log[]> {
+    const { RPC } = getRPCConstants(chainId)
+    if (!RPC[0]) throw new Error('Unknown chain id.')
+    const res = await fetch(RPC[0], {
```

Comment:
I'm a little doubt if we should do a direct fetch here. Maybe we should let the `Web3Context` provide a `fetch`, or even further, let `Web3Context` provide this functionality.

Reply:
The metamask RPC is not reliable for method 'eth_getLogs', since it is set casually by the user. So we use our own default controllable RPC.So we use our own default controllable RPC.

Reply:
I moved `getPastLogs()` to the `EthereumServices/network.ts`